src/components/Rating.jsx

```diff
@@ -0,0 +1,136 @@
+    updatePseudoClassStyle(this.ratingStyleId, style.ratingStyleBefore);
+    updatePseudoClassStyle(this.ratingHolderStyleId, style.ratingHolderStyleBefore);
+  }
+
+  /**
+   * Funtion to remove pseudo classes from dom once component is removed.
+   */
+  componentWillUnmount() {
+    removeStyle(this.ratingStyleId);
+    removeStyle(this.ratingHolderStyleId);
+  }
+
+  /**
+   * Calculate width of highlighted stars, the function uses this.state.hoverRating
+   * if it exists else it uses this.state.rating.
+   */
+  _getWidth() {
+    var value = this.state.hoverRating?this.state.hoverRating:this.state.rating;
+    return (Math.round(value) * 20) + '%';
+  }
+
+  /**
+   * When mouse hover overs the component this function will set the hoverRating
+   * in the component state depending on mouse position.
+   */
+  _onMouseMove(e) {
+    if(!this.props.disabled) {
+      const wrapperNode = React.findDOMNode(this.refs.holder);
+      const holderWidth = wrapperNode.getBoundingClientRect().width;
+      const mouseMoved = e.pageX - wrapperNode.getBoundingClientRect().left;
+      const newRating = mouseMoved * 5 / holderWidth;
+      this.setState({
+        hoverRating: newRating
+      });
+    }
+  }
+
+  /**
+   * Function will reset hover rating as mouse leaves.
+   */
+  _onMouseLeave() {
+    if(!this.props.disabled) {
+      this.setState({
+        hoverRating: undefined
+      });
+    }
+  }
+
+  /**
+   * The function will update rating when component is clicked.
+   */
+  _onClick() {
+    if(!this.props.disabled) {
+      this.setState({
+        rating: this.state.hoverRating
+      });
+    }
+  }
+
+  //We may not allow custom styling to component since that can interfere with the rating calculations.
+  /**
+   * Function to render component.
+   */
+  render () {
+    const width = this._getWidth();
+    const ratingCalculatedStyle = extend({}, style.ratingStyle, {width: width});
+
+    return <div ref="holder"
+                style={style.ratingHolderStyle}
+                className={ this.ratingHolderStyleId }
+                onMouseMove={ this._onMouseMove.bind(this) }
+                onMouseLeave={ this._onMouseLeave.bind(this) }
+                onClick={ this._onClick.bind(this) }>
+                <div style={ratingCalculatedStyle}
+                  className={ this.ratingStyleId }></div>
+              </div>;
+  }
+}
+
+/**
+ * Props of Rating component
+ */
+Rating.propTypes = {
+  value: React.PropTypes.oneOf([0, 1, 2, 3, 4, 5]).isRequired,
+  disabled: React.PropTypes.bool
+};
+
+/**
+ * Setting default prop values.
+ */
+Rating.defaultProps = {
+  disabled: false
+};
+
+Rating.displayName = 'Belle Rating';
+
+/**
+ * Function to create pseudo classes for styles.
+ */
+function updatePseudoClassStyle(styleId, style) {
+  const styles = [
+    {
+      id: styleId,
+      style: style,
+      pseudoClass: ':before'
+    }
+  ];
+  injectStyles(styles);
```

Comment:
Ideally this should not be called multiple times as it triggers a DOM manipulation. I would recommend to update the `updatePseudoClassStyle` & take both styleIDs